src/main/java/radiounit/EricssonLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,193 @@
+		FrequencyBand band = carrier.getCarrierFrequencies();
+		boolean isLte = false;
+		
+		for (LteFrequencyBand lteFreq : LteFrequencyBand.values())
+		{
+			if (band == lteFreq)
+			{
+				isLte = true;
+				break;
+			}
+		}
+
+		if (!isLte)
+		{
+			System.err.println("Cannot add WCDMA carrier to LTE radio!");
+			return;
+		}
+		
+		if (getCarrierById(carrier.getCarrierId()) != null)
+		{
+			System.err.println("Carrier with id " + carrier.getCarrierId() + " already exists!");
+			return;
+		}
+		
+		addCarrierInternal(carrier);
 	}
 
 	public void signalScalingEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] signalScalingEricssonLte");
 	}
 
 	public void modifyCarrierEricssonLte(Integer carrierId, FrequencyBand frequencyBand) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] modifyCarrierEricssonLte: " + carrierId + ", " + frequencyBand.getBand());
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		Carrier existingCarrier = getCarrierById(carrierId);
+		
+		removeCarrierInternal(idx);
+		
+		existingCarrier.setFrequencyBand(frequencyBand);
+		
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierEricssonLte(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] removeCarrierEricssonLte: " + carrierId);
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
+		{
+			System.err.println("Carrier with id " + carrierId + " does not exist!");
+			return;
+		}
+		
+		removeCarrierInternal(idx);
 	}
 
 	public void selfDiagnosticsEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] selfDiagnosticsEricssonLte");
 	}
 
-	public void removeAllCharactersEricssonLte() {
-		// TODO Auto-generated method stub
-
+	public void removeAllCarriersEricssonLte() {
+		System.out.println("[EricssonLteRadioUnitReceiver] removeAllCarriersEricssonLte");
+		
+		for (int i = 0; i < getAllCarriersInternal().size(); ++i)
+		{
+			removeCarrierInternal(i);
+		}
 	}
 
 	public List<Carrier> getCarriers() {
-		return carriers;
+		System.out.println("[EricssonLteRadioUnitReceiver] getCarriers");
+		
+		List<Carrier> carrierList = getAllCarriersInternal();
+		
+		for (Carrier c : carrierList)
+		{
+			System.out.println("Carrier: " + c);
+		}
+		
+		return carrierList;
+	}
+	
+	private Carrier getCarrierById(int id)
+	{
+		int idx = getCarrierIdxById(id);
+		
+		if (idx == -1)
+		{
+			return null;
+		}
+		
+		return getAllCarriersInternal().get(idx);
+	}
+	
+	private int getCarrierIdxById(int id)
+	{
+		for (int i = 0; i < getAllCarriersInternal().size(); ++i)
+		{
+			Carrier c = getAllCarriersInternal().get(i);
+			
+			if (c.getCarrierId() == id)
+			{
+				return i;
+			}
+		}
+		
+		return -1;
```

Comment:
Ah, so does -1 represent non-existent carriers?

Reply:
Correct - it indicates that the carrier doesn't exist in the list, kind of like index "-1" doesn't exist in an array/list